src/lib/components/forms/form/form-components/text-input/text-input.component.spec.ts

```diff
@@ -0,0 +1,157 @@
+    });
+
+    it('should disable the input when #isDisabled is set', async () => {
+        expect(await input.isDisabled()).toBe(false);
+        component.isDisabled = true;
+        fixture.detectChanges();
+
+        expect(await input.isDisabled()).toBe(true);
+    });
+
+    it('should mark the input readonly when #isReadonly is set', async () => {
+        expect(await input.isReadonly()).toBe(false);
+        component.isReadonly = true;
+
+        fixture.detectChanges();
+        expect(await input.isReadonly()).toEqual(true);
+    });
+
+    it('should have #name as label of the input', async () => {
+        const formField: MatFormFieldHarness = await loader.getHarness(MatFormFieldHarness);
+        expect(await formField.getLabel()).toBe('');
+
+        component.name = 'My Label';
+        fixture.detectChanges();
+        expect(await formField.getLabel()).toBe(component.name);
+    });
+
+    it('should have iban validator when isIban', async () => {
+
+        const validIBAN: string = 'DE02370501980001802057';
+        const invalidIBAN: string = 'DE0237050198000180205'; // last number 7 is removed
+
+        const formField: MatFormFieldHarness = await loader.getHarness(MatFormFieldHarness);
+        expect(component.isIban).toBe(false);
+        component.isIban = true;
+        component.writeValue(validIBAN)
+        fixture.detectChanges();
+
+        expect(await formField.isControlValid()).toBeTruthy();
+
+        component.writeValue(invalidIBAN)
+        fixture.detectChanges();
+
+        expect(await formField.isControlValid()).toBeFalsy();
+    });
+
+    describe('with tooltip', () => {
+        let tooltip: MockTooltipDirective;
+        beforeEach(() => {
+            tooltip = fixture.debugElement.query(By.directive(MockTooltipDirective)).injector.get(MockTooltipDirective);
+        });
+
+        it('should apply the tcTooltip directive to the form field with the given #tooltipText', () => {
+            expect(tooltip).toBeTruthy();
+            expect(tooltip.tcTooltip).toBe('');
+
+            component.tooltipText = 'My tooltip';
+            fixture.detectChanges();
+
+            expect(tooltip.tcTooltip).toBe(component.tooltipText);
+        });
+
+        it(`should set the tooltip's placement according to the input #tooltipPlacement`, () => {
+            expect(tooltip.placement).toBe(TerraPlacementEnum.TOP);
+
+            component.tooltipPlacement = TerraPlacementEnum.BOTTOM;
+            fixture.detectChanges();
+
+            expect(tooltip.placement).toBe(component.tooltipPlacement);
+        });
+    });
+
+    it('should update the value of the input when writing a new value via `writeValue()`', async () => {
+        const value: string = 'my test string';
+        component.writeValue(value);
+        expect(component.value).toEqual(value);
+
+        fixture.detectChanges();
+        expect(await input.getValue()).toEqual(value.toString());
```

Comment:
```suggestion
        expect(await input.getValue()).toEqual(value);
```
value ist doch schon ein string^^

Reply:
jap 🙉